### PR TITLE
API-379/API-380: create and list family variants

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**
Add create family variant in the API
Add list family variants in the API

Bonus:
Fix integration tests with invalid data.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/FamilyVariant/ListFamilyVariantIntegration.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                            "a_yes_no"
+                        ],
+                        "axes" : [
+                            "a_yes_no"
+                        ]
+                    }
+                ],
+                "_links" : {
+                    "self" : {
+                        "href" : "http://localhost/api/rest/v1/families/familyVariantA1"
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/FamilyVariant/ListFamilyVariantIntegration.php]: should be `http://localhost/api/rest/v1/families/familyA/variants/familyVariantA1` no ?
(same remark for all your tests)

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/FamilyVariant/ListFamilyVariantIntegration.php [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    {
+        $client = $this->createAuthenticatedClient();
+
+        $client->request('GET', 'api/rest/v1/families/familyA/variants');
+
+        $expected = <<<JSON
+{
+    "_links" : {
+        "first" : {
+            "href" : "http://localhost/api/rest/v1/families?page=1&limit=10&with_count=false"
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/FamilyVariant/ListFamilyVariantIntegration.php]: should be `http://localhost/api/rest/v1/families/familyA/variants?page=1&limit=10&with_count=false`

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/FamilyVariant/ListFamilyVariantIntegration.php (hunk 1 above)
Comment (ahocquard) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/FamilyVariant/ListFamilyVariantIntegration.php]: nice catch, fixed

**Comment by ahocquard** [comment]
CI green, just failing due to "variant group" fails (existing on master too)

**Pull request merged**